tst/EnergyPlus/unit/VariableSpeedCoils.unit.cc

```diff
@@ -6963,4 +6963,160 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_Coil_Defrost_Power_Fix_Test)
+        "    5.68,                    !- Speed 5 Reference Unit Air Flow Rate At Rated Conditions {m3/s}",
+        "    773.3,                   !- Speed 5 2017 Rated Evaporator Fan Power Per Volume Flow Rate",
+        "    934.4,                   !- Speed 5 2023 Rated Evaporator Fan Power Per Volume Flow Rate",
+        "    16.36,                   !- Speed 5 Reference Unit Condenser Flow Rate at Rated Conditions {m3/s}",
+        "    ,                        !- Speed 5 Reference Unit Pad Effectiveness of Evap Precooling at Rated Conditions {dimensionless}",
+        "    CapacityCurve,           !- Speed 5 Total Cooling Capacity Function of Temperature Curve Name",
+        "    CAPFF Curve,             !- Speed 5 Total Cooling Capacity Function of Air Flow Fraction Curve Name",
+        "    PowerCurve,              !- Speed 5 Energy Input Ratio Function of Temperature Curve Name",
+        "    EIRFF Curve;             !- Speed 5 Energy Input Ratio Function of Air Flow Fraction Curve Name",
+
+        "Curve:Biquadratic,",
+        "    CapacityCurve, 1, 0, 0, 0, 0, 0, 10, 25.5, 7.2, 48.8, , , Temperature, Temperature, Dimensionless;",
+        "Curve:Biquadratic,",
+        "    PowerCurve, 1, 0, 0, 0, 0, 0, 10, 25.5, 7.2, 48.8, , , Temperature, Temperature, Dimensionless;",
+        "Curve:Cubic,",
+        "    CAPFF Curve, 1, 0, 0, 0, 0, 1, , , Dimensionless, Dimensionless;",
+        "Curve:Cubic,",
+        "    EIRFF Curve, 1, 0, 0, 0, 0, 1, , , Dimensionless, Dimensionless;",
+        "Curve:Quadratic,",
+        "    PLF Curve, 0.85, 0.8333, 0.0, 0.0, 0.3, 0.85, 1.0, Dimensionless, Dimensionless;",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    // get coil inputs
+    VariableSpeedCoils::GetVarSpeedCoilInput(*state);
+    // Setting predefined tables is needed though
+    OutputReportPredefined::SetPredefinedTables(*state);
+    // Set up some environmental parameters
+    state->dataEnvrn->OutDryBulbTemp = 5.0;
+    state->dataEnvrn->OutHumRat = 0.0009;
+    state->dataEnvrn->OutBaroPress = 99000.0;
+    state->dataEnvrn->OutWetBulbTemp =
+        Psychrometrics::PsyTwbFnTdbWPb(*state, state->dataEnvrn->OutDryBulbTemp, state->dataEnvrn->OutHumRat, state->dataEnvrn->OutBaroPress);
+    state->dataEnvrn->WindSpeed = 5.0;
+    state->dataEnvrn->WindDir = 270.0;
+    state->dataEnvrn->StdRhoAir = 1.1;
+    // set coil parameters
+    int const CyclingScheme = DataHVACGlobals::ContFanCycCoil;
+    int DXCoilNum = 1;
+    DataHVACGlobals::CompressorOperation CompressorOp = DataHVACGlobals::CompressorOperation::Off;
+    int constexpr SpeedCal = 1;
+    Real64 SensLoad = 0.0;
+    Real64 LatentLoad = 0.0;
+    Real64 PartLoadFrac = 0.0;
+    Real64 OnOffAirFlowRatio = 1.0;
+    Real64 SpeedRatio = 0.0;
+
+    // run coil init
+    VariableSpeedCoils::InitVarSpeedCoil(*state, DXCoilNum, SensLoad, LatentLoad, CyclingScheme, OnOffAirFlowRatio, SpeedRatio, SpeedCal);
```

Comment:
This throws a floating point exception (when enabled) at https://github.com/NREL/EnergyPlus/blob/eb6665f392fc51eb6727ec5d8f7af21722941d79/src/EnergyPlus/VariableSpeedCoils.cc#L5559

Reply:
If you protect against it, and set AirMassFlowRatio to 1.0 in case it's zero, it'll still throw at

https://github.com/NREL/EnergyPlus/blob/eb6665f392fc51eb6727ec5d8f7af21722941d79/src/EnergyPlus/VariableSpeedCoils.cc#L8620-L8625

TotCap is zero, so InletEnthalpy == hADP